server/graphql/v1/mutations/orders.js

```diff
@@ -96,8 +99,29 @@ async function checkOrdersLimit(order, remoteUser, reqIp) {
   }
 }
 
+const checkGuestContribution = order => {
+  const { interval, guestInfo } = order;
+
+  if (!parseToBoolean(config.guestContributions.enable)) {
+    throw new Error('Guest contributions are not enabled yet');
+  } else if (interval) {
+    throw new Error('You need to sign up to create a recurring contribution');
+  } else if (!guestInfo) {
+    throw new Error('You need to provide a guest profile with an email for logged out contributions');
+  } else if (guestInfo.email && !isEmail(guestInfo.email)) {
+    throw new Error('You need to provide a valid email');
+  } else if (!guestInfo.email && !guestInfo.token) {
+    throw new Error('When contributing as a guest, you either need to provide an email or a token');
+  } else if (order.totalAmount > 5000) {
+    if (!guestInfo.name || !guestInfo.location?.address || !guestInfo.location.country) {
+      throw new Error('Contributions that are more than $5000 must have an address attached');
+    }
+  }
+};
+
 async function checkRecaptcha(order, remoteUser, reqIp) {
-  if (['ci', 'test'].includes(config.env)) {
+  // Disabled for all environments
+  if (['ci', 'test', 'development', 'production'].includes(config.env)) {
```

Comment:
I just noticed this and didn't realise we could include special return cases for CI! 🤯

Reply:
It's better to not have conditional behaviors based on the env when possible, for reproducibility and testability. But sometimes it's very convenient!